dask/array/core.py

```diff
@@ -2249,6 +2258,120 @@ def to_zarr(arr, url, component=None, storage_options=None,
+
+        :param array: tiledb.Array
+        :return: list
+        """
+        schema = array.schema
+        return list(schema.domain.dim(i).tile for i in range(schema.ndim))
+
+    import tiledb
+    tiledb_config = storage_options or dict()
+    key = tiledb_config.pop('key', None)
+
+    if isinstance(uri, tiledb.Array):
+        tdb = uri
+    else:
+        tdb = tiledb.open(uri, attr=attribute, config=tiledb_config, key=key)
+
+    if tdb.schema.sparse:
+        raise Exception("Sparse TileDB arrays are not supported")
+
+    if not attribute:
+        if tdb.schema.nattr > 1:
+            raise TypeError("keyword 'attribute' must be provided"
+                            "when loading a multi-attribute TileDB array")
+        else:
+            attribute = tdb.schema.attr(0).name
+
+    chunks = tiledb_to_chunks(tdb)
+
+    assert(len(chunks) == tdb.schema.ndim)
+
+    return from_array(tdb, chunks, name='tiledb-%s' % uri)
+
+
+def to_tiledb(darray, uri, compute=True, return_stored=False,
+              storage_options=None, **kwargs):
+    """Save 'array' using the TileDB storage manager, to any TileDB-supported URI,
+    including local disk, S3, or HDFS.
+
+    See https://docs.tiledb.io for more information about TileDB.
+
+    Parameters
+    ----------
+
+    darray: dask.array
+        A dask array to write.
+    uri:
+        Any supported TileDB storage location.
+    storage_options:
+        Dict containing any configuration options for the TileDB backend.
+        see https://docs.tiledb.io/en/stable/tutorials/config.html
+    compute, return_stored: see ``store()``
+    :return:
+        None
+    """
+    import tiledb
+
+    tiledb_config = storage_options or dict()
+    # encryption key, if any
+    key = tiledb_config.pop('key', None)
+
+    if not isinstance(darray, Array):
+        raise TypeError("to_tiledb first argument must be a dask array")
+
+    if not _check_regular_chunks(darray.chunks):
+        raise ValueError('Attempt to save array to TileDB with irregular '
+                         'chunking, please call `arr.rechunk(...)` first.')
+
+    if isinstance(uri, str):
+        chunks = [c[0] for c in darray.chunks]
+        key = kwargs.pop('key', None)
+        # create a suitable, empty, writable TileDB array
+        tdb = tiledb.empty_like(uri, darray, tile=chunks, config=tiledb_config, key=key)
+    elif isinstance(uri, tiledb.Array):
+        tdb = uri
+        # sanity checks
+        if not ((darray.shape == tdb.shape) and
+                (darray.dtype == tdb.dtype) and
+                (darray.ndim == tdb.ndim)):
+            raise ValueError("Target TileDB array layout is not compatible with "
+                             "source array")
+    else:
+        raise ValueError("'uri' must be string pointing to supported TileDB store location "
+                         "or an open, writable TileDB array.")
+
+    if not tdb.isopen and tdb.iswritable:
```

Comment:
shouldn't it be `not (tdb.isopen and tdb.iswritable)` ?

Reply:
good catch, thanks